app/models/activity.rb

```diff
@@ -60,19 +60,25 @@ def schedule=(ice_cube_schedule)
     super(ice_cube_schedule.to_yaml)
   end
 
-  def to_time_ranges
-    schedule.occurrences_between(plan.start_date.beginning_of_day, plan.end_date.end_of_day).map do |o|
-      TimeRange.new(
-        start_time: o.start_time,
-        end_time: o.end_time,
-        value: o.duration / 60, # Duration in minutes.
-        user_id: plan.user_id
-      )
+  def to_time_ranges # rubocop:disable Metrics/AbcSize
+    Rails.cache.fetch(time_ranges_cache_key, expires_in: 1.week) do
+      schedule.occurrences_between(plan.start_date.beginning_of_day, plan.end_date.end_of_day).map do |o|
+        TimeRange.new(
+          start_time: o.start_time,
+          end_time: o.end_time,
+          value: o.duration / 60, # Duration in minutes.
+          user_id: plan.user_id
+        )
+      end
     end
   end
 
   private
 
+  def time_ranges_cache_key
+    "Activity#to_time_ranges##{id}##{updated_at}"
```

Comment:
Recommend `updated_at.to_f` here because the default is only accurate to the second and can lead to tricky bugs, e.g. 2 updates could happen within 1 second. Also makes a shorter key.

```
irb(main):029:0" "#{Plan.first.updated_at}"
=> "2021-01-29 13:52:43 UTC"
irb(main):030:0> "#{Plan.first.updated_at.to_f}"
=> "1611928363.130215"
```

Reply:
thx